### ISSUE TITLE
Flatpak generated sh files do not run on Steam Deck

### ISSUE THREAD

**Issue by outphase** [redromnon/HeroicBashLauncher#19]
I am using 2.2.0-Flatpak. I manually added the sh files to Steam because they were not automatically added (Issue #18). 

Running them in the Deck UI fails after pressing the Play button.

Running the sh file from Dolphin does not show any actions.

Running the sh file from by right clicking and running in terminal gives the error: 

`bwrap: execvp ./HeroicBashLauncher: No such file or directory`

**Comment by outphase** [comment]
> Did you run from inside the GameFiles folder?

Yes. I did it once with Konsole manually navigating to GameFiles and once with right click "Run In Konsole"

**Comment by redromnon** [comment]
Alright. Can you open the terminal in GameFiles and then run the executable `./HeroicBashLauncher`?

**Comment by outphase** [comment]
```
(deck@deckphase GameFiles)$ ./HeroicBashLauncher
Checking if Zenity is installed:
3.32.0

Cleaning left over game files if any...
sh: line 1: /home/deck/Games/HeroicBashLauncher-2.2.0_Flatpak/GameFiles/binaries/legendary: No such file or directory


Done! Now creating launch files for your Epic Games library ...

Death's Door [65d73e3be8824829b5b788bd849b6559]...

Disco Elysium [7334aba246154b63857435cb9c7eecd5]...

The Jackbox Party Pack 6 [Astrapia]...

The Jackbox Party Pack 5 [Geranium]...

A Short Hike [d6407c9e6fd54cb492b8c6635480d792]...

Rise of the Tomb Raider: 20 Year Celebration [f7cc1c999ac146f39b356f53e3489514]...


...Process finished. Launch files stored in GameFiles folder.
Have fun gaming!
```

**Comment by outphase** [comment]
Yes, the scripts have been created. 

No, they do not launch the game from the terminal.

```
(deck@deckphase GameFiles)$ ./DeathsDoor_Heroic.sh
bwrap: execvp ./HeroicBashLauncher: No such file or directory
```

**Comment by redromnon** [comment]
HeroicBashLauncher is right there, and I cannot understand why it can't find it.

**Comment by outphase** [comment]
I'm guessing the flatpak or bwrap commands can't see into that folder where it resides

**Comment by TomMora96** [comment]
Hello, I was experiencing the same issue in Linux Mint, after some research (i'm not an expert) I found that flatpak needs to grant HGL access to the file directory where HeroicBashLaucher is located to be able to execute it. Using 
`sudo flatpak override com.heroicgameslauncher.hgl --filesystem=/home/tomardo/Downloads/HeroicBashLauncher`
I was able to run the game script.
However, it failed to launch the game, instead this was shown in the console
```
(base) tomardo@tomy-linux:~/Downloads/HeroicBashLauncher/GameFiles$ ./EuropaUniversalisIV_Heroic.sh
Checking if Zenity is installed:
Gtk-Message: 21:49:17.260: Failed to load module "xapp-gtk3-module"
3.32.0
Launch command:
STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/tomardo/.steam/steam STEAM_COMPAT_DATA_PATH='/home/tomardo/Games/Heroic/Prefixes/Europa-Universalis-IV' /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch da0103e959e54d139d0c109ded3b3672 --no-wine --wrapper "'/home/tomardo/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-1/bin/wine' run" 
[cli] INFO: Logging in...
[Core] INFO: Trying to re-use existing login session...
[cli] INFO: Checking for updates...
[Core] INFO: Getting authentication token...
[cli] INFO: Launching da0103e959e54d139d0c109ded3b3672...
wine: created the configuration directory '/home/tomardo/.wine'
wineserver: using server-side synchronization.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
002c:fixme:actctx:parse_depend_manifests Could not find dependent assembly L"Microsoft.Windows.Common-Controls" (6.0.0.0)
(base) tomardo@tomy-linux:~/Downloads/HeroicBashLauncher/GameFiles$ 0048:fixme:actctx:parse_depend_manifests Could not find dependent assembly L"Microsoft.Windows.Common-Controls" (6.0.0.0)
0050:fixme:actctx:parse_depend_manifests Could not find dependent assembly L"Microsoft.Windows.Common-Controls" (6.0.0.0)
0048:err:ole:StdMarshalImpl_MarshalInterface Failed to create ifstub, hr 0x80004002
0048:err:ole:CoMarshalInterface Failed to marshal the interface {6d5140c1-7436-11ce-8034-00aa006009fa}, hr 0x80004002
0048:err:ole:apartment_get_local_server_stream Failed: 0x80004002
0050:err:ole:StdMarshalImpl_MarshalInterface Failed to create ifstub, hr 0x80004002
0050:err:ole:CoMarshalInterface Failed to marshal the interface {6d5140c1-7436-11ce-8034-00aa006009fa}, hr 0x80004002
0050:err:ole:apartment_get_local_server_stream Failed: 0x80004002
005c:err:ole:com_get_class_object class {ff48dba4-60ef-4201-aa87-54103eef594e} not registered
005c:err:ole:com_get_class_object no class object {ff48dba4-60ef-4201-aa87-54103eef594e} could be created for context 0x1
005c:err:tabtip:create_uia_event_handler Failed to create IUIAutomation interface, hr 0x80040154
0048:err:explorer:start_tabtip_process Unexpected termination of tabtip.exe - exit code -1
0048:err:ole:start_rpcss Failed to open RpcSs service
0064:err:ole:com_get_class_object class {ff48dba4-60ef-4201-aa87-54103eef594e} not registered
0064:err:ole:com_get_class_object no class object {ff48dba4-60ef-4201-aa87-54103eef594e} could be created for context 0x1
0064:err:tabtip:create_uia_event_handler Failed to create IUIAutomation interface, hr 0x80040154
0050:err:explorer:start_tabtip_process Unexpected termination of tabtip.exe - exit code -1
0050:fixme:imm:ImeSetActiveContext (0x4203d0, 0): stub
0050:fixme:imm:ImmReleaseContext (0000000000010020, 00000000004203D0): stub
002c:fixme:imm:ImeSetActiveContext (0x420210, 1): stub
002c:fixme:imm:ImmReleaseContext (0000000000010050, 0000000000420210): stub
0098:err:setupapi:SetupDefaultQueueCallbackW copy error 1812 L"@C:\\windows\\system32\\drivers\\wineusb.sys,-1" -> L"C:\\windows\\inf\\wineusb.inf"
00a0:fixme:file:NtLockFile I/O completion on lock not implemented yet
00a0:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
00a0:err:mscoree:LoadLibraryShim error reading registry key for installroot
00a0:err:mscoree:LoadLibraryShim error reading registry key for installroot
00a0:err:mscoree:LoadLibraryShim error reading registry key for installroot
00a0:err:mscoree:LoadLibraryShim error reading registry key for installroot
00a8:fixme:file:NtLockFile I/O completion on lock not implemented yet
00a8:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
00a8:err:mscoree:LoadLibraryShim error reading registry key for installroot
00a8:err:mscoree:LoadLibraryShim error reading registry key for installroot
00a8:err:mscoree:LoadLibraryShim error reading registry key for installroot
00a8:err:mscoree:LoadLibraryShim error reading registry key for installroot
00a8:fixme:msi:internal_ui_handler internal UI not implemented for message 0x0b000000 (UI level = 5)
00a8:fixme:msi:internal_ui_handler internal UI not implemented for message 0x0b000000 (UI level = 5)
00a0:fixme:msi:internal_ui_handler internal UI not implemented for message 0x0b000000 (UI level = 1)
00a0:fixme:msi:internal_ui_handler internal UI not implemented for message 0x0b000000 (UI level = 1)
wine: configuration in L"/home/tomardo/.wine" has been updated.
Application could not be started, or no application associated with the specified file.
ShellExecuteEx failed: File not found.
```

I'm not sure what's all that about, it looks like most of it is from wine. I'm able to launch the game from HGL, but not from the .sh file.

**Comment by redromnon** [comment]
Looks like it's a wine issue for me too. Can you compare the Launch command with the one in the game's log in HGL? Maybe even try other games too.

**Comment by outphase** [comment]
Here is the complete log from when I ran a game until I reached the main menu

```
Launch Command: STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/deck/.var/app/com.heroicgameslauncher.hgl/.steam/steam STEAM_COMPAT_DATA_PATH='/home/deck/Games/Heroic/Prefixes/Deaths-Door' /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch 65d73e3be8824829b5b788bd849b6559 --no-wine --wrapper "'/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton' run"
        System Info:

  Heroic Version: 2.2.6 Oden
  Legendary Version:  0.20.25 Our Benefactors
  OS: Freedesktop.org KERNEL: 5.13.0-valve10.1-1-neptune-02144-g7fffaf925dfb ARCH: x64
  CPU: AMD Custom APU 0405 @2.03 GOVERNOR: schedutil
  RAM: Total: 15.6 GB Available: 13.3 GB
  GRAPHICS: GPU0: Device 163f VRAM: 512MB DRIVER:
  PROTOCOL: x11
Game Settings: {
    "checkUpdatesInterval": 10,
    "enableUpdates": false,
    "addDesktopShortcuts": false,
    "addStartMenuShortcuts": false,
    "autoInstallDxvk": false,
    "autoInstallVkd3d": false,
    "checkForUpdatesOnStartup": true,
    "customWinePaths": [],
    "defaultInstallPath": "/home/deck/Games/Heroic",
    "defaultWinePrefix": "/home/deck/Games/Heroic/Prefixes",
    "language": "en",
    "maxWorkers": 0,
    "minimizeOnLaunch": true,
    "nvidiaPrime": false,
    "otherOptions": "",
    "showUnrealMarket": false,
    "showFps": false,
    "useGameMode": false,
    "userInfo": {
        "epicId": "4544e4ab6b964e2b8d4967e446b4d958",
        "name": "deck"
    },
    "wineCrossoverBottle": "Heroic",
    "winePrefix": "/home/deck/Games/Heroic/Prefixes/Deaths-Door",
    "wineVersion": {
        "bin": "'/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton'",
        "name": "Proton - Proton 7.0",
        "type": "proton"
    },
    "altLegendaryBin": "",
    "altGogdlBin": "",
    "audioFix": false
    "darkTrayIcon": false,
    "discordRPC": false,
    "egsLinkedPath": "",
    "enableEsync": false,
    "enableFsync": false,
    "exitToTray": false,
    "maxRecentGames": 5,
    "offlineMode": false,
    "showMangohud": false,
    "startInTray": false,
    "autoSyncSaves": true,
    "maxSharpness": 2,
    "savesPath": "/home/deck/Games/Heroic/Prefixes/Deaths-Door/pfx/drive_c/users/steamuser/AppData/LocalLow/Acid Nerve/DeathsDoor/SAVEDATA/",
    "targetExe": "",
    "useSteamRuntime": false
}

        Game Log:
        [cli] INFO: Logging in...
[Core] INFO: Trying to re-use existing login session...
[cli] INFO: Checking for updates...
[Core] INFO: Getting authentication token...
[cli] INFO: Launching 65d73e3be8824829b5b788bd849b6559...
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
WARNING: radv is not a conformant Vulkan implementation, testing use only.
WARNING: radv is not a conformant Vulkan implementation, testing use only.
WARNING: radv is not a conformant Vulkan implementation, testing use only.
WARNING: radv is not a conformant Vulkan implementation, testing use only.
WARNING: radv is not a conformant Vulkan implementation, testing use only.
WARNING: radv is not a conformant Vulkan implementation, testing use only.
```

**Comment by redromnon** [comment]
> Here is the complete log from when I ran a game until I reached the main menu
> 
> ```
> Launch Command: STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/deck/.var/app/com.heroicgameslauncher.hgl/.steam/steam STEAM_COMPAT_DATA_PATH='/home/deck/Games/Heroic/Prefixes/Deaths-Door' /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch 65d73e3be8824829b5b788bd849b6559 --no-wine --wrapper "'/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton' run"
>         System Info:
> 
>   Heroic Version: 2.2.6 Oden
>   Legendary Version:  0.20.25 Our Benefactors
>   OS: Freedesktop.org KERNEL: 5.13.0-valve10.1-1-neptune-02144-g7fffaf925dfb ARCH: x64
>   CPU: AMD Custom APU 0405 @2.03 GOVERNOR: schedutil
>   RAM: Total: 15.6 GB Available: 13.3 GB
>   GRAPHICS: GPU0: Device 163f VRAM: 512MB DRIVER:
>   PROTOCOL: x11
> Game Settings: {
>     "checkUpdatesInterval": 10,
>     "enableUpdates": false,
>     "addDesktopShortcuts": false,
>     "addStartMenuShortcuts": false,
>     "autoInstallDxvk": false,
>     "autoInstallVkd3d": false,
>     "checkForUpdatesOnStartup": true,
>     "customWinePaths": [],
>     "defaultInstallPath": "/home/deck/Games/Heroic",
>     "defaultWinePrefix": "/home/deck/Games/Heroic/Prefixes",
>     "language": "en",
>     "maxWorkers": 0,
>     "minimizeOnLaunch": true,
>     "nvidiaPrime": false,
>     "otherOptions": "",
>     "showUnrealMarket": false,
>     "showFps": false,
>     "useGameMode": false,
>     "userInfo": {
>         "epicId": "4544e4ab6b964e2b8d4967e446b4d958",
>         "name": "deck"
>     },
>     "wineCrossoverBottle": "Heroic",
>     "winePrefix": "/home/deck/Games/Heroic/Prefixes/Deaths-Door",
>     "wineVersion": {
>         "bin": "'/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton'",
>         "name": "Proton - Proton 7.0",
>         "type": "proton"
>     },
>     "altLegendaryBin": "",
>     "altGogdlBin": "",
>     "audioFix": false
>     "darkTrayIcon": false,
>     "discordRPC": false,
>     "egsLinkedPath": "",
>     "enableEsync": false,
>     "enableFsync": false,
>     "exitToTray": false,
>     "maxRecentGames": 5,
>     "offlineMode": false,
>     "showMangohud": false,
>     "startInTray": false,
>     "autoSyncSaves": true,
>     "maxSharpness": 2,
>     "savesPath": "/home/deck/Games/Heroic/Prefixes/Deaths-Door/pfx/drive_c/users/steamuser/AppData/LocalLow/Acid Nerve/DeathsDoor/SAVEDATA/",
>     "targetExe": "",
>     "useSteamRuntime": false
> }
> 
>         Game Log:
>         [cli] INFO: Logging in...
> [Core] INFO: Trying to re-use existing login session...
> [cli] INFO: Checking for updates...
> [Core] INFO: Getting authentication token...
> [cli] INFO: Launching 65d73e3be8824829b5b788bd849b6559...
> fsync: up and running.
> wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
> WARNING: radv is not a conformant Vulkan implementation, testing use only.
> WARNING: radv is not a conformant Vulkan implementation, testing use only.
> WARNING: radv is not a conformant Vulkan implementation, testing use only.
> WARNING: radv is not a conformant Vulkan implementation, testing use only.
> WARNING: radv is not a conformant Vulkan implementation, testing use only.
> WARNING: radv is not a conformant Vulkan implementation, testing use only.
> ```

I was actually referring to @TomMora96 
But even you can compare if the entire launch command at the top matches when you run the .sh file.

**Comment by outphase** [comment]
I added /home/deck/Games/HeroicBashLauncher/GameFiles into HGL's permissions via flatseal and ran the .sh again.

Now, when I run the .sh, it does stuff, but the game will not launch. Nothing happens after this part of the log, and HGL doesn't have a log of the game ever being run.

```
Checking if Zenity is installed:
Gtk-Message: 22:47:49.165: Failed to load module "canberra-gtk-module"
3.32.0
Launch command:
STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/deck/.steam/steam STEAM_COMPAT_DATA_PATH='/home/deck/Games/Heroic/Prefixes/Rise-of-the-Tomb-Raider-20-Year-Celebration' /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch f7cc1c999ac146f39b356f53e3489514 --no-wine --wrapper "'/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton' run" 
[Core] INFO: Trying to re-use existing login session...
[cli] INFO: Got 10 remote save game(s)
[cli] INFO: Checking "Rise of the Tomb Raider: 20 Year Celebration" (f7cc1c999ac146f39b356f53e3489514)
[cli] INFO: Overriding save path with "/home/deck/Games/Heroic/Prefixes/Rise-of-the-Tomb-Raider-20-Year-Celebration/pfx/drive_c/users/steamuser/My Documents/Rise of the Tomb Raider/4544e4ab6b964e2b8d4967e446b4d958/"...
[cli] INFO: Local save for "Rise of the Tomb Raider: 20 Year Celebration" is newer
[cli] INFO: - Cloud save date: N/A
[cli] INFO: - Local save date: 2022-03-15 03:19:09
[cli] INFO: Uploading local savegame...
[SGH] INFO: Packing savegame for "f7cc1c999ac146f39b356f53e3489514", input folder: /home/deck/Games/Heroic/Prefixes/Rise-of-the-Tomb-Raider-20-Year-Celebration/pfx/drive_c/users/steamuser/My Documents/Rise of the Tomb Raider/4544e4ab6b964e2b8d4967e446b4d958/
[SGH] INFO: Chunk #1 "ChunksV4/61/AAA5E44B732A9A73_F7E9DC41D939403887C5A5B6891FF836.chunk" created
[SGH] INFO: Manifest "manifests/2022.03.15-03.19.09.manifest" written (409 bytes)
[Core] INFO: Starting upload...
[Core] INFO: Finished uploading savegame.
[cli] INFO: Logging in...
[Core] INFO: Trying to re-use existing login session...
[cli] INFO: Checking for updates...
[Core] INFO: Getting authentication token...
[cli] INFO: Launching f7cc1c999ac146f39b356f53e3489514...
```

**Comment by redromnon** [comment]
So I can confirm a bug here. I'm sure this is the reason why the games aren't launching (those with Proton)
`STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/deck/.steam/steam` is supposed to be `STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/deck/.var/app/com.heroicgameslauncher.hgl/.steam/steam`.

I'll also try to fix the "executable not found" issue too.

**Comment by redromnon** [comment]
So I've fixed the bugs in the latest release v2.2.1
